### PR TITLE
Sort `--generate-rcfile` ouput

### PR DESCRIPTION
**Pull request by cdce8p**
## Description
Sort output of `--generate-rcfile` similar to  the  one from `--generate-toml-config` and  update  example pylintrc.
Noticed this while trying the release workflow.

Note: To make backporting easier, don't squash merge this PR.

https://github.com/PyCQA/pylint/blob/15dd079792ce8e6b37856450f9e4a928f11b3549/pylint/config/arguments_manager.py#L704-L707

### CODE REVIEW

File: pylint/config/arguments_manager.py [hunk 1]
```diff
@@ -447,7 +447,10 @@ def generate_config(
         )
         options_by_section = {}
         sections = []
-        for group in self._arg_parser._action_groups:
+        for group in sorted(
+            self._arg_parser._action_groups,
+            key=lambda x: (x.title != "Main", x.title),
+        ):
```
Comment (Pierre-Sassoulas) [inline pylint/config/arguments_manager.py:453]: Shouldn't the options also be sorted line 460 to 464 ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
Makes perfect sense, thank you.

File: pylint/config/arguments_manager.py (hunk 1 above)
Comment (cdce8p) [reply pylint/config/arguments_manager.py:453]: 👍🏻  Done, although it didn't change much. I force pushed the changes to keep the history clean for the merge commit.

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**